rpc/legacyrpc/methods.go

```diff
@@ -205,63 +176,36 @@ type lazyHandler func() (interface{}, *dcrjson.RPCError)
 // returning a closure that will execute it with the (required) wallet and
 // (optional) consensus RPC server.  If no handlers are found and the
 // chainClient is not nil, the returned handler performs RPC passthrough.
-func lazyApplyHandler(request *dcrjson.Request, w *wallet.Wallet,
-	chainClient *dcrrpcclient.Client, l *loader.Loader) lazyHandler {
-
-	handlerData, ok := rpcHandlers[request.Method]
-	if ok && handlerData.handlerWithChain != nil && w != nil && chainClient != nil {
-		return func() (interface{}, *dcrjson.RPCError) {
-			cmd, err := dcrjson.UnmarshalCmd(request)
-			if err != nil {
-				return nil, dcrjson.ErrRPCInvalidRequest
-			}
-			resp, err := handlerData.handlerWithChain(cmd, w, chainClient)
-			if err != nil {
-				return nil, jsonError(err)
-			}
-			return resp, nil
-		}
-	}
-	if ok && handlerData.handlerWithLoader != nil && w != nil {
-		return func() (interface{}, *dcrjson.RPCError) {
-			cmd, err := dcrjson.UnmarshalCmd(request)
-			if err != nil {
-				return nil, dcrjson.ErrRPCInvalidRequest
-			}
-			resp, err := handlerData.handlerWithLoader(cmd, w, l)
-			if err != nil {
-				return nil, jsonError(err)
-			}
-			return resp, nil
-		}
-	}
-	if ok && handlerData.handler != nil && w != nil {
+func lazyApplyHandler(s *Server, request *dcrjson.Request) lazyHandler {
+	handlerData, ok := handlers[request.Method]
+	if !ok {
 		return func() (interface{}, *dcrjson.RPCError) {
-			cmd, err := dcrjson.UnmarshalCmd(request)
-			if err != nil {
-				return nil, dcrjson.ErrRPCInvalidRequest
+			chainClient, ok := s.requireChainClient()
+			if !ok {
+				return nil, &dcrjson.RPCError{
```

Comment:
Let's define a package error variable for this condition, similar to ErrUnloadedWallet.